firmware/user/modes/mode_flight.c

```diff
@@ -0,0 +1,861 @@
+           0,  36,     48,   3,     36,  48,     36,  45,     45,  48,     15,  48,     45,  15,      0,  45,   
+          54,  39,      6,  54,     54,  51,     51,  39,      9,  51,     54,   9,     51,   0,     60,  51,   
+           9,  60,     60,  57,     57,  51,     12,  57,     60,  12,     57,   0,     63,  57,     12,  63,   
+          63,  45,     45,  57,     63,  15,     69,   3,     48,  69,     48,  66,     66,  69,     30,  69,   
+          66,  30,     15,  66,     75,   6,     42,  75,     42,  72,     72,  75,     18,  75,     72,  18,   
+           3,  72,     81,   9,     54,  81,     54,  78,     78,  81,     21,  81,     78,  21,      6,  78,   
+          87,  12,     60,  87,     60,  84,     84,  87,     24,  87,     84,  24,      9,  84,     93,  15,   
+          63,  93,     63,  90,     90,  93,     27,  93,     90,  27,     12,  90,     96,  69,     30,  96,   
+          96,  72,     72,  69,     96,  18,     99,  75,     18,  99,     99,  78,     78,  75,     99,  21,   
+         102,  81,     21, 102,    102,  84,     84,  81,    102,  24,    105,  87,     24, 105,    105,  90,   
+          90,  87,    105,  27,    108,  93,     27, 108,    108,  66,     66,  93,    108,  30,    114,  18,   
+          96, 114,     96, 111,    111, 114,     33, 114,    111,  33,     30, 111,    117,  21,     99, 117,   
+          99, 114,    114, 117,     33, 117,    120,  24,    102, 120,    102, 117,    117, 120,     33, 120,   
+         123,  27,    105, 123,    105, 120,    120, 123,     33, 123,    108, 111,    108, 123,    123, 111,   
+        };
+
+void ICACHE_FLASH_ATTR DrawGeoSphere()
+{
+    int i;
+    int nrv = sizeof(indices)/sizeof(uint16_t);
+    for( i = 0; i < nrv; i+=2 )
+    {
+        int16_t * c1 = &verts[indices[i]];
+        int16_t * c2 = &verts[indices[i+1]];
+        Draw3DSegment( c1, c2 );
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+static void ICACHE_FLASH_ATTR flightGameUpdate( flight_t * flight )
+{
+    uint8_t bs = flight->buttonState;
+
+    // First clear the OLED
+    clearDisplay();
+
+
+    char framesStr[8] = {0};
+    ets_snprintf(framesStr, sizeof(framesStr), "%d", flight->buttonState);
+    plotText(0, 0, framesStr, TOM_THUMB, WHITE);
+
+    if( bs & 1 ) flight->hpr[0]++;
+    if( bs & 4 ) flight->hpr[0]--;
+
+    static int ij;
+    ij++;
+    SetupMatrix();
+
+    tdRotateEA( ProjectionMatrix, -20, 0, 0 );
+    tdRotateEA( ModelviewMatrix, ij, 0, 0 );
+    //tdTranslate( ModelviewMatrix, 0, 0, 200 );
+
+    PIN_FUNC_SELECT( PERIPHS_IO_MUX_U0TXD_U, 3); //Set to GPIO.  
```

Comment:
Overclocking seems both useful and opaque. Can you make `EnableOverclock()` and `DisableOverclock()` functions and put them in the `utils` folder somewhere, so that other modes can use them if necessary?

Reply:
Ok

Reply:
`OVERCLOCK_SECTION_ENABLE()` and `OVERCLOCK_SECTION_DISABLE()` are pretty nice. Why are you toggling GPIO1(TX), just to do time measurements on a logic analyzer? If that's the case, I'd recommend macro-ing that too. I know it's not your modus operandi, but I try to separate the hardware management from the modes, so that contributors don't need to know every in and out of the ESP to do useful things. You could also redefine the macro for the emulator to measure system time between the calls and print that too.

Reply:
100% Logic analyzer, to do really precise timing measurements.  It's easier for me to see the things in a waterfall view on the analyzer. This pin control will be undone in time.